### PR TITLE
Rename compression level to compression stage

### PR DESCRIPTION
**Pull request by vanyalzr**

### CODE REVIEW

File: nncf/api/compression.py [hunk 1]
```diff
@@ -130,33 +130,33 @@ def get_state(self) -> Dict[str, object]:
...
     NONE = 0
     PARTIAL = 1
     FULL = 2
 
-    def __add__(self, other: 'CompressionLevel') -> 'CompressionLevel':
+    def __add__(self, other: 'CompressionStage') -> 'CompressionStage':
         """
-        Defines compression level of a composite compression controller, consist of
-        two algorithms, where `self` is the compression level of the first algorithm
-        and other - compression level of the second one.
```
Comment (ljaljushkin) [inline nncf/api/compression.py:146]: These names made sense for me with compression level class.
full/partial compression level 

but with stages probably we should come up with other names, like
FINAL, INTERMEDIATE

Comment (vshampor) [reply nncf/api/compression.py:146]: `to distinguish between uncompressed, partially compressed, and fully compressed models`

UNCOMPRESSED, PARTIALLY_COMPRESSED, FULLY_COMPRESSED

Comment (ljaljushkin) [reply nncf/api/compression.py:146]: > `to distinguish between uncompressed, partially compressed, and fully compressed models`
> 
> UNCOMPRESSED, PARTIALLY_COMPRESSED, FULLY_COMPRESSED

yes, this one is better

**Review by alexsu52 — APPROVED** [review]

File: nncf/api/compression.py (hunk 1 above)
Comment (vanyalzr) [reply nncf/api/compression.py:146]: agreed, done

**Review by vshampor — APPROVED** [review]

**Pull request merged**